en/21/06.md

```diff
@@ -0,0 +1,199 @@
+
+You would call this function like so from within trait / struct that implements it:
+
+```
+self.eat_hamburgers(100);
+```
+
+Like this from outside the trait that implements it:
+
+```
+Person::eat_hamburgers(100);
+```
+
+Or like this from outside the struct that implements it:
+
+```
+person.eat_hamburgers(100);
+```
+
+## Creating a structure type object
+
+In Rust creating a structure type object is done very easy:
+
+```
+let given_name = ManagedBuffer::from(b"Bob");
+let given_age = 30u32;
+let person = Person { name: given_name, age: given_age };
+```
+
+If for example you have a variable whose name is matching the structure field's than the syntax can be simplified:
+
+
+```
+let name = ManagedBuffer::from(b"Bob");
+let age = 30u32;
+let person = Person { name, age };
+```
+## Using an indexed mapper
+
+Adding an element to an indexed mapper is done just like adding the an element to the normal one, the only difference being the requirement of the index
+
+
+```
+self.persons(id).set(new_person);
+
+```
+## Reading the value of a mapper
+
+In order to read the value of a `singleValueMapper` we will use the method `get`:
+
+
+```
+let my_person = self.persons(id).get();
+
+```
+## Updating a mapper value
+
+For updating the value of a mapper we can use the `update` method which expects a closure as parameter. You can see the closure as a no name function which we use to access or to process elements in an advaned way within a data type.
+
+```
+let name = ManagedBuffer::from(b"Joe");
+let age = 35u32;
+let new_person = Person { name, age };
+
+self.persons(id).update(|stored_person| *stored_person = new_person );
+```
+Here between `||` we will have the storage type which we can access it under what name we waht, following the operation we want to do to it. Inside the closure we can use variables declared outside. 
+
+> Notice: stored_person is a `&Person` (reference to a Person) reason why in order to access its value we need to dereference it with `*`
+
+Note that you can make a more complex closure in which you can make more operations. An example of it would be:
+
+```
+let new_number = 42u32;
+let mut old_favorite_number = 0u32;
+
+self.favorite_number(id).update(|number| { 
+  old_favorite_number = *number;
+  *number = new_number;
+});
+```
+
+# Put it to the test
+
+In our app, we're going to need to be able to create some zombies. Let's create a function for that.
+The id of the introduced zombie should be stored in a separate storage of type `SingleValueMapper<usize>` called `zombie_count` that increases every time a new zombie is being added. 
+2. Create a function named `create_zombie`. It should take two parameters: **\_name** (a `ManagedBuffer`), and **\_dna** (a `u64`).
+
+3. Inside the body the first line  we will write the update for `zombies_count` which will access its value as `id`. Here id doesn't need to be dereferenced, since we access it as `&usize` for the index of `zombies`.
```

Comment:
Please consider rephrasing this. It's not clear what the student is expected to do.